lib/storage/file.js

```diff
@@ -1173,6 +1173,8 @@ File.prototype.getSignedUrl = function(options, callback) {
   }[options.action];
 
   var name = encodeURIComponent(this.name);
+  var targetGeneration = is.undefined(this.generation) ?
+    false : encodeURIComponent(this.generation);
```

Comment:
Also, I think `generation` only comes in the form of integers, `encodeURIComponent` might not be necessary here.